test/test_ekfac.py

```diff
@@ -0,0 +1,1094 @@
+        [(X, y)],
+    )
+    with raises(ValueError, match="reduction"):
+        ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+
+    # create new EKFAC with different model and try to load state dict
+    wrong_model = Sequential(Linear(D_in, 10), ReLU(), Linear(10, D_out))
+    wrong_params = list(wrong_model.parameters())
+    ekfac_new = EKFACLinearOperator(
+        wrong_model,
+        MSELoss(reduction="sum"),
+        wrong_params,
+        [(X, y)],
+    )
+    with raises(RuntimeError, match="loading state_dict"):
+        ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+
+    # create new EKFAC and load state dict
+    ekfac_new = EKFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+        check_deterministic=False,  # turn off to avoid computing EKFAC again
+    )
+    ekfac_new.load_state_dict(load(EKFAC_PATH, weights_only=False))
+    # clean up
+    os.remove(EKFAC_PATH)
+
+    # check that the two EKFACs are equal
+    compare_state_dicts(ekfac.state_dict(), ekfac_new.state_dict())
+    test_vec = rand(ekfac.shape[1])
+    report_nonclose(ekfac @ test_vec, ekfac_new @ test_vec)
+
+
+def test_from_state_dict():
+    """Test that EKFACLinearOperator can be created from state dict."""
+    manual_seed(0)
+    batch_size, D_in, D_out = 4, 3, 2
+    X = rand(batch_size, D_in)
+    y = rand(batch_size, D_out)
+    model = Linear(D_in, D_out)
+
+    params = list(model.parameters())
+    # create and compute EKFAC
+    ekfac = EKFACLinearOperator(
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+    )
+
+    # save state dict
+    state_dict = ekfac.state_dict()
+
+    # create new EKFAC from state dict
+    kfac_new = EKFACLinearOperator.from_state_dict(state_dict, model, params, [(X, y)])
+
+    # check that the two EKFACs are equal
+    compare_state_dicts(ekfac.state_dict(), kfac_new.state_dict())
+    test_vec = rand(ekfac.shape[1])
+    report_nonclose(ekfac @ test_vec, kfac_new @ test_vec)
+
+
+@mark.parametrize("fisher_type", EKFACLinearOperator._SUPPORTED_FISHER_TYPE)
+@mark.parametrize(
+    "separate_weight_and_bias", [True, False], ids=["separate_bias", "joint_bias"]
+)
+@mark.parametrize(
+    "exclude", [None, "weight", "bias"], ids=["all", "no_weights", "no_biases"]
+)
+@mark.parametrize("shuffle", [False, True], ids=["", "shuffled"])
+def test_ekfac_closer_to_exact_than_kfac(
+    inv_case,
+    shuffle: bool,
+    exclude: str,
+    separate_weight_and_bias: bool,
+    fisher_type: FisherType,
+):
+    """Test that EKFAC is closer in Frobenius norm to the exact quantity than KFAC."""
+    model, loss_func, params, data, batch_size_fn = inv_case
+    params = maybe_exclude_or_shuffle_parameters(params, model, exclude, shuffle)
+
+    # Compute exact block-wise ground truth quantity.
+    linop = (
+        EFLinearOperator if fisher_type == FisherType.EMPIRICAL else GGNLinearOperator
```

Comment:
What if the Fisher type is type-I MC?

In this case, EKFAC should be a better approximation to the MC Fisher than KFAC, and so the linop (the 'exact' quantity) would have to be `FisherMCLinearOperator` with the same seed as (E)KFAC.

Reply:
Then we could always set `mc_samples=1` below.

Reply:
That's a great idea, thanks!

Reply:
Should be refactored like [here](https://github.com/f-dangel/curvlinops/pull/173/files#diff-7a31260cb833244158c8ec94421e00372de13b3d7fca3ccc37785e21647ba73cR897-R901), no?

Reply:
I actually also tried to use the same trick here, but it doesn't work. Maybe the eigendecompositions are not stable enough for these cases?

Reply:
Did you pass the same `seed` to `FisherMCLinearOperator` than to `EKFACLinerOperator` and `KFACLinearOperator`?

Reply:
Yes, I did. I noticed that only a single case was failing and modified it to have 3 instead of 2 classes (it was a CE loss case). After this adjustment, it seems stable. See the refactor [here](https://github.com/f-dangel/curvlinops/pull/173/commits/b1a20748011a3b861fa121dcff237c4f3dce36b8).